edgecore panic when run on raspberrypi 4

<!-- Please use this template while reporting a bug and provide as much info as possible. Thanks!-->


**What happened**:
i compile edgecore on   raspberrypi 4 , and run it:
```
2019-08-26 15:49:19.977 +08:00 INFO client/client_plugins.go:24 Install client plugin, protocol: rest
2019-08-26 15:49:19.977 +08:00 INFO registry/discovery.go:18 Installed service discovery plugin: edge
I0826 15:49:19.978469   11826 sql.go:20] Begin to register twin model
I0826 15:49:19.978610   11826 db.go:37] DB meta for module twin has been registered
I0826 15:49:19.978757   11826 db.go:37] DB meta for module twin has been registered
I0826 15:49:19.978959   11826 db.go:37] DB meta for module twin has been registered
2019-08-26 15:49:19.978 +08:00 INFO core/module.go:41 module twin registered
panic: interface conversion: interface {} is int64, not int

goroutine 1 [running]:
github.com/kubeedge/kubeedge/edge/pkg/edged.getConfig(0xc97470)
	/root/projects/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:392 +0xc84
github.com/kubeedge/kubeedge/edge/pkg/edged.newEdged(0x0, 0xb, 0x5989fc0)
	/root/projects/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:429 +0x18
github.com/kubeedge/kubeedge/edge/pkg/edged.Register()
	/root/projects/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:264 +0x14
github.com/kubeedge/kubeedge/edge/cmd/edgecore/app.registerModules()
	/root/projects/src/github.com/kubeedge/kubeedge/edge/cmd/edgecore/app/server.go:46 +0x18
github.com/kubeedge/kubeedge/edge/cmd/edgecore/app.NewEdgeCoreCommand.func1(0x5a76dc0, 0x3a987e8, 0x0, 0x0)
	/root/projects/src/github.com/kubeedge/kubeedge/edge/cmd/edgecore/app/server.go:34 +0x14
github.com/kubeedge/kubeedge/vendor/github.com/spf13/cobra.(*Command).execute(0x5a76dc0, 0x58a00e0, 0x0, 0x0, 0x5a76dc0, 0x58a00e0)
	/root/projects/src/github.com/kubeedge/kubeedge/vendor/github.com/spf13/cobra/command.go:830 +0x204
github.com/kubeedge/kubeedge/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x5a76dc0, 0x58b6000, 0x200e2cc, 0x58000e0)
	/root/projects/src/github.com/kubeedge/kubeedge/vendor/github.com/spf13/cobra/command.go:914 +0x234
github.com/kubeedge/kubeedge/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/root/projects/src/github.com/kubeedge/kubeedge/vendor/github.com/spf13/cobra/command.go:864
main.main()
	/root/projects/src/github.com/kubeedge/kubeedge/edge/cmd/edgecore/edgecore.go:18 +0x44

```

the code which trigger panic is 
```
	conf.memoryCapacity = config.CONFIG.GetConfigurationByKey("edged.edged-memory-capacity-bytes").(int)
	if conf.memoryCapacity == 0 {
		conf.memoryCapacity = MinimumEdgedMemoryCapacity
	}

```



**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- KubeEdge version: master
- Hardware configuration: raspberrypi 4
- OS (e.g. from /etc/os-release):
 os image: https://www.raspberrypi.org/downloads/raspbian/

- Kernel (e.g. `uname -a`):
- Others:
cpu info:
```
lscpu
Architecture:        armv7l
Byte Order:          Little Endian
CPU(s):              4
On-line CPU(s) list: 0-3
Thread(s) per core:  1
Core(s) per socket:  4
Socket(s):           1
Vendor ID:           ARM
Model:               3
Model name:          Cortex-A72
Stepping:            r0p3
CPU max MHz:         1500.0000
CPU min MHz:         600.0000
BogoMIPS:            108.00
Flags:               half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32

```

goland info ： `go1.12.9.linux-armv6l.tar.gz `



cc @sids-b @kevin-wangzefeng @fisherxu @subpathdev

--- a/edge/pkg/edged/edged.go
+++ b/edge/pkg/edged/edged.go
@@ -238,7 +238,7 @@ type Config struct {
 	nodeName                  string
 	nodeNamespace             string
 	interfaceName             string
-	memoryCapacity            int
+	memoryCapacity            int64
 	nodeStatusUpdateInterval  time.Duration
 	devicePluginEnabled       bool
 	gpuPluginEnabled          bool
@@ -389,7 +389,16 @@ func getConfig() *Config {
 		conf.clusterDomain = ""
 	}
 
-	conf.memoryCapacity = config.CONFIG.GetConfigurationByKey("edged.edged-memory-capacity-bytes").(int)
+	//Deal with 32-bit and 64-bit compatibility issues: issue #1070
+	switch v := config.CONFIG.GetConfigurationByKey("edged.edged-memory-capacity-bytes").(type) {
+	case int:
+		conf.memoryCapacity = int64(v)
+	case int64:
+		conf.memoryCapacity = v
+	default:
+		panic("Invalid type for edged.edged-memory-capacity-bytes, valid types are one of [int,int64].")
+	}
+
 	if conf.memoryCapacity == 0 {
 		conf.memoryCapacity = MinimumEdgedMemoryCapacity
 	}

--- a/edge/test/integration/scripts/execute.sh
+++ b/edge/test/integration/scripts/execute.sh
@@ -38,7 +38,8 @@ if pgrep edgecore >/dev/null
 then
     echo "edgecore process is Running"
 else
-    echo "edgecore process is not started"
+    echo "edgecore process is not started, log info:"
+    cat edgecore.log
     exit 1
 fi
 PWD=${curpath}/test/integration

